### PR TITLE
Add slams

### PR DESCRIPTION
**Pull request by wrefgtzweve**

### CODE REVIEW

File: lua/autorun/server/sv_limit_explosives.lua [hunk 1]
```diff
@@ -1,7 +1,12 @@
 local limits = {
-    m9k_proxy = 4
+    m9k_proxy = 4,
+    npc_satchel = 15
 }
 
+for _, k in ipairs( player.GetHumans() ) do
+    k.placedExplosives = nil
+end
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_limit_explosives.lua]: I don't think there will be any players when this file is loaded?

File: lua/autorun/server/sv_limit_explosives.lua [hunk 2]
```diff
@@ -11,7 +16,10 @@ local function getExplosivesTable( ply, class )
     return ply.placedExplosives[class]
 end
 
-local function onExplosiveCreated( owner, ent )
+local function onExplosiveCreated( ent )
+    if not IsValid( ent ) then return end
+    if not limits[ent:GetClass()] then return end
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_limit_explosives.lua]: Could you move the `class` variable above this line and use it in the check?

File: lua/autorun/server/sv_limit_explosives.lua [hunk 3]
```diff
@@ -28,25 +36,26 @@ local function onExplosiveCreated( owner, ent )
 
 end
 
-local function onExplosiveRemoved( owner, ent )
+local function onExplosiveRemoved( ent )
+    if not IsValid( ent ) then return end
+    if not limits[ent:GetClass()] then return end
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_limit_explosives.lua]: Similar comment about `class` here (maybe call it `entClass` too)

File: lua/autorun/server/sv_limit_explosives.lua [hunk 4]
```diff
@@ -28,25 +36,26 @@ local function onExplosiveCreated( owner, ent )
...
 
     local explosives = getExplosivesTable( owner, ent:GetClass() )
     if explosives == nil then return end
+
     table.RemoveByValue( explosives, ent )
 end
 
 hook.Remove( "OnEntityCreated", "CFC_LimitExplosives" )
 hook.Add( "OnEntityCreated", "CFC_LimitExplosives", function( ent )
-    if limits[ent:GetClass()] == nil then return end
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_limit_explosives.lua:42]: I think it's still safe to check the class here

**Review by sarahsturgeon — CHANGES_REQUESTED** [review]

File: lua/autorun/server/sv_limit_explosives.lua (hunk 1 above)
Comment (wrefgtzweve) [reply lua/autorun/server/sv_limit_explosives.lua]: oh mb that was for testing fml

File: lua/autorun/server/sv_limit_explosives.lua (hunk 4 above)
Comment (wrefgtzweve) [reply lua/autorun/server/sv_limit_explosives.lua:42]: it wasnt because it's ran on a 0 tick timer. for some reason crossbow bolts error it idk why

**Comment by wrefgtzweve** [comment]
ready

File: lua/autorun/server/sv_limit_explosives.lua (hunk 4 above)
Comment (sarahsturgeon) [reply lua/autorun/server/sv_limit_explosives.lua:42]: The 0-tick timer makes sure a lot of stuff on the entity is setup (cppi owner and such), but the class _must_ be defined on created

Comment (wrefgtzweve) [reply lua/autorun/server/sv_limit_explosives.lua:42]:
```
[cfc_placed_explosives_limiter] addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua:16: Tried to use a NULL entity!
  1. GetClass - [C]:-1
   2. onExplosiveCreated - addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua:16
    3. unknown - addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua:55

Timer Failed! [Simple][@addons/cfc_placed_explosives_limiter/lua/autorun/server/sv_limit_explosives.lua (line 54)]
```

**Review by sarahsturgeon — APPROVED** [review]
Looks good. Might be worthwhile to localize `IsValid` at the top of this file. It's used a lot, and very frequently in here.

**Pull request merged**